src/sagemaker/cli/framework_upgrade.py

```diff
@@ -0,0 +1,284 @@
+        short_version (str): Abbreviated framework version (e.g. 1.0, 1.5).
+        full_version (str): Complete framework version (e.g. 1.0.0, 1.5.2).
+        image_type (str): Framework image type, it could be "training", "inference" or "eia".
+        scopes (str): Framework image type, it could be "training", "inference".
+        processors (str): Supported processors (e.g. "cpu,gpu").
+        py_versions (str): Supported Python versions (e.g. "py3,py37").
+        tag_prefix (str): Algorithm image's tag prefix.
+     """
+    py_versions = py_versions.split(",")
+    processors = processors.split(",")
+    latest_registries, latest_py_versions, latest_repository = _get_latest_values(
+        framework, existing_content, image_type
+    )
+    if not py_versions:
+        py_versions = latest_py_versions
+
+    if framework in DLC_FRAMEWORK:
+        add_dlc_framework_version(
+            existing_content,
+            short_version,
+            full_version,
+            image_type,
+            processors,
+            py_versions,
+            latest_registries,
+            latest_repository,
+        )
+    else:
+        scopes = scopes.split(",")
+        add_algo_version(
+            existing_content,
+            processors,
+            scopes,
+            full_version,
+            py_versions,
+            latest_registries,
+            latest_repository,
+            tag_prefix,
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Framework upgrade tool.")
+    parser.add_argument("--framework", help="Name of the framework (e.g. tensorflow, mxnet, etc.)")
+    parser.add_argument("--short-version", help="Abbreviated framework version (e.g. 2.0)")
+    parser.add_argument("--full-version", help="Full framework version (e.g. 2.0.1)")
+    parser.add_argument("--image-type", help="Framework image type (e.g. training, inference, eia)")
+    parser.add_argument("--processors", help="Suppoted processors (e.g. cpu, gpu)")
+    parser.add_argument("--py-versions", help="Supported Python versions (e.g. py3,py37)")
+    parser.add_argument(
+        "--scopes", help="Scopes for the Algorithm image (e.g. inference, training)"
+    )
+    parser.add_argument(
+        "--tag-prefix", help="Tag prefix of the Algorithm image (e.g. ray-0.8.5-torch)"
+    )
+    parser.add_argument("--region", help="New region to be added (e.g. us-west-2)")
+    parser.add_argument("--account", help="Registry account of new region")
+
+    args = parser.parse_args()
+    framework = args.framework
+    short_version = args.short_version
+    full_version = args.full_version
+    image_type = args.image_type
+    processors = args.processors
+    py_versions = args.py_versions
+    scopes = args.scopes
+    tag_prefix = args.tag_prefix
+    region = args.region
+    account = args.account
+
+    if not framework:
+        raise ValueError("Please specify a framework or algorithm name to upgrade")
```

Comment:
you can do `required=True` when setting up the arg parser to avoid having to do this separately

https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument